clients/extension/scripts/popup.js

```diff
@@ -5,58 +5,20 @@
  */
 'use strict';
 
-/** @typedef {typeof import('./extension-entry.js') & {console: typeof console}} BackgroundPage */
-/** @typedef {import('../../../lighthouse-core/lib/lh-error.js')} LighthouseError */
+/** @typedef {import('./extension-controller.js').Settings} Settings */
 
-/**
- * Error strings that indicate a problem in how Lighthouse was run, not in
- * Lighthouse itself, mapped to more useful strings to report to the user.
- */
-const NON_BUG_ERROR_MESSAGES = {
-  'DNS_FAILURE': 'DNS servers could not resolve the provided domain.',
-  'INVALID_URL': 'Lighthouse can only audit URLs that start' +
-      ' with http:// or https://.',
-
-  // chrome extension API errors
-  'multiple tabs': 'You probably have multiple tabs open to the same origin. ' +
-      'Close the other tabs to use Lighthouse.',
-  // The extension debugger API is forbidden from attaching to the web store.
-  // @see https://chromium.googlesource.com/chromium/src/+/5d1f214db0f7996f3c17cd87093d439ce4c7f8f1/chrome/common/extensions/chrome_extensions_client.cc#232
-  'The extensions gallery cannot be scripted': 'The Lighthouse extension cannot audit the ' +
-      'Chrome Web Store. If necessary, use the Lighthouse CLI to do so.',
-  'Cannot access a chrome': 'The Lighthouse extension cannot audit ' +
-      'Chrome-specific urls. If necessary, use the Lighthouse CLI to do so.',
-  'Cannot access contents of the page': 'Lighthouse can only audit URLs that start' +
-      ' with http:// or https://.',
-};
-
-const subpageVisibleClass = 'subpage--visible';
-
-/** @type {?URL} */
-let siteURL = null;
-/** @type {boolean} */
-let isRunning = false;
-
-function getLighthouseVersion() {
-  return chrome.runtime.getManifest().version;
-}
-
-function getLighthouseCommitHash() {
-  return '__COMMITHASH__';
-}
+/** @type {import('./extension-controller.js')} */
+const ExtensionController = (() => {
+  // @ts-ignore: for popup-test.js
+  if (window.ControllerMock) return window.ControllerMock;
+  return require('./extension-controller.js');
+})();
 
-function getChromeVersion() {
-  // @ts-ignore
-  return /Chrome\/([0-9.]+)/.exec(navigator.userAgent)[1];
-}
+// `update_url` only exists in production - it's a property that the Chrome web store injects.
+const DEV = !('update_url' in chrome.runtime.getManifest());
```

Comment:
> once this is merged I imagine we'll want it hitting real viewer by default even for dev anyway when we just make tweaks to extension :)

You're right, since the viewer is live now this is OK to be removed.